### PR TITLE
add changelog for v5.0.0

### PR DESCRIPTION
**Pull request by roycaihw**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+* Add support for client-side spam filtering of events
+
+   * [https://github.com/kubernetes/kubernetes/pull/47367](https://github.com/kubernetes/kubernetes/pull/47367)
+
+* Support http etag and caching
+
+   * [https://github.com/kubernetes/kubernetes/pull/50404](https://github.com/kubernetes/kubernetes/pull/50404)
+
+* Extend SetHeader Requests method ito accept multiple values
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you remove this entry? It's to trivial to be mentioned in changelog.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/50639](https://github.com/kubernetes/kubernetes/pull/50639)
+
+* Add a heap to client-go
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you change it to "Adding priority queue support to informer cache."? That's the main use case.

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/49752](https://github.com/kubernetes/kubernetes/pull/49752)
+
+* Add openstack auth provider
+
+   * [https://github.com/kubernetes/kubernetes/pull/39587](https://github.com/kubernetes/kubernetes/pull/39587)
+
+* Make it possible to allow discovery errors for controllers
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you remove this one?

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+* Adds metrics for checking reflector health.
+
+   * [https://github.com/kubernetes/kubernetes/pull/48224](https://github.com/kubernetes/kubernetes/pull/48224)
+
+* Client-go now includes the leaderelection package
+
+   * [https://github.com/kubernetes/kubernetes/pull/39173](https://github.com/kubernetes/kubernetes/pull/39173)
+
+* OpenAPI downloads protobuf rather than Json to improve performance
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you move it to the "breaking change" section?

Could you also update the text to "The discovery client now fetches the protobuf encoded OpenAPI schema and returns `openapi_v2.Document`"?

File: CHANGELOG.md [hunk 5]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/46803](https://github.com/kubernetes/kubernetes/pull/46803)
+
+**API changes:**
+
+* Autoscaling v2alpha1 graduated to v2beta1
+
+   * [https://github.com/kubernetes/kubernetes/pull/50708](https://github.com/kubernetes/kubernetes/pull/50708)
+
+* Copy list expansions for DaemonSet, Deployment, ReplicaSet, and StatefulSet to apps/v1beta2 from extensions/v1beta1 and apps/v1beta1
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you remove this entry?

File: CHANGELOG.md [hunk 6]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/50871](https://github.com/kubernetes/kubernetes/pull/50871)
+
+* Add ControllerRevision to apps/v1beta2
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you remove this entry?

File: CHANGELOG.md [hunk 7]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/50698](https://github.com/kubernetes/kubernetes/pull/50698)
+
+* Promote CronJobs to batch/v1beta1
+
+   * [https://github.com/kubernetes/kubernetes/pull/41901](https://github.com/kubernetes/kubernetes/pull/41901)
+
+* Promote rbac.authorization.k8s.io/v1beta1 to rbac.authorization.k8s.io/v1 with no changes. The rbac.authorization.k8s.io/v1alpha1 version is deprecated and will be removed in a future release.
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Maybe just "Promote rbac.authorization.k8s.io/v1beta1 to rbac.authorization.k8s.io/v1"

File: CHANGELOG.md [hunk 8]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/49642](https://github.com/kubernetes/kubernetes/pull/49642)
+
+* Enforce explicit references to API group client interfaces in clientsets to avoid ambiguity.
```
Comment (caesarxuchao) [inline CHANGELOG.md]: This is not API change, could you move it to the "breaking change" section?

File: CHANGELOG.md [hunk 9]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/49370](https://github.com/kubernetes/kubernetes/pull/49370)
+
+* The 504 timeout error was returning a JSON error body that indicated it was a 500. The body contents now correctly report a 500 error.
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you remove this entry?

File: CHANGELOG.md [hunk 10]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/49678](https://github.com/kubernetes/kubernetes/pull/49678)
+
+* Add a new API object apps/v1beta2.ReplicaSet
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you remove this entry?

File: CHANGELOG.md [hunk 11]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/49238](https://github.com/kubernetes/kubernetes/pull/49238)
+
+* Adding type apps/v1beta2.DaemonSet
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you remove this entry?

File: CHANGELOG.md [hunk 12]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/49071](https://github.com/kubernetes/kubernetes/pull/49071)
+
+* Add a new API version apps/v1beta2
+
+   * [https://github.com/kubernetes/kubernetes/pull/48746](https://github.com/kubernetes/kubernetes/pull/48746)
+
+* Add PriorityClass API object under new "scheduling" API group
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Add a new API version scheduling/v1alpha1

File: CHANGELOG.md [hunk 13]
```diff
@@ -15,6 +15,98 @@ be exported in other repos. The issue is tracked
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/48377](https://github.com/kubernetes/kubernetes/pull/48377)
+
+* The generic RESTClient type (`k8s.io/client-go/rest`) no longer exposes `LabelSelectorParam` or `FieldSelectorParam` methods - use `VersionedParams` with `metav1.ListOptions` instead. The `UintParam` method has been removed. The `timeout` parameter will no longer cause an error when using `Param()`.
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Could you move this entry to breaking changes?

File: CHANGELOG.md (hunk 1 above)
Comment (roycaihw) [reply CHANGELOG.md]: Updated and squashed

File: CHANGELOG.md [hunk 14]
```diff
@@ -2,18 +2,77 @@ TODO: This document was manually maintained so might be incomplete. The
...
+
+   * [https://github.com/kubernetes/kubernetes/pull/48377](https://github.com/kubernetes/kubernetes/pull/48377)
+
+**Breaking changes:**
+
+* pkg/api and pkg/apis are moved to [k8s.io/api](https://github.com/kubernetes/api). Other kubernetes repositories also import types from there, so they are composable with client-go.
+
+* Helper functions in pkg/api and pkg/apis are also removed. They are planned to be exported in other repos. The issue is tracked [here](https://github.com/kubernetes/kubernetes/issues/48209#issuecomment-314537745). During the transition, you'll have to copy the helper functions to your projects.
+
+* The discovery client now fetches the protobuf encoded OpenAPI schema and returns `openapi\_v2.Document`
```
Comment (caesarxuchao) [inline CHANGELOG.md]: Remove the "\"

**Comment by caesarxuchao** [comment]
lgtm. Thanks a lot! I'll create the tag and then merge.

**Pull request merged**